### PR TITLE
feat(query): ownership cover stage/udf

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

ownership cover stage and udf

Fixes https://github.com/datafuselabs/databend/issues/14211

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14217)

### CODE REVIEW

**Comment by TCeason** [comment]
After this pr merged, we can delete setting: enable_experimental_rbac_check.

File: src/meta/app/src/principal/user_grant.rs [hunk 1]
```diff
@@ -21,19 +21,31 @@ use enumflags2::BitFlags;
 use crate::principal::UserPrivilegeSet;
 use crate::principal::UserPrivilegeType;
 
-/// [`GrantObjectByID`] is used to maintain the grant object by id. Using ID over name
+/// [`OwnerObject`] is used to maintain the grant object that support rename by id. Using ID over name
```
Comment (flaneur2020) [inline src/meta/app/src/principal/user_grant.rs:24]: OwnerObject -> OwnershipObject ?

OwnerObject might have a bit of confuse about it represents the `owner`, but `OwnerObject` not targets an owner, but the object of an ownership.

**Comment by flaneur2020** [comment]
all the others are LGTM :rocket:

File: src/meta/app/src/principal/user_grant.rs (hunk 1 above)
Comment (TCeason) [reply src/meta/app/src/principal/user_grant.rs]: Done

**Review by flaneur2020 — APPROVED** [review]

**Review by everpcpc — APPROVED** [review]

**Pull request merged**